backend/s3/ibm_signer.go

```diff
@@ -16,20 +16,23 @@ type Authenticator interface {
 }
 
 // IbmIamSigner is a structure for signing requests using IBM IAM.
-// Requeres APIKey and Resource InstanceID
+// Requires APIKey and Resource InstanceID
 type IbmIamSigner struct {
 	APIKey     string
 	InstanceID string
 	Auth       Authenticator
 }
+// IamEndpoint set the private IAM endpoint which is accessible on public and private clsters
+// If not set, IamAuthenticator sets defualt value i.e., public IAM endpoint which is not accessible on private clusters
+const IamEndpoint = "https://private.iam.cloud.ibm.com"
```

Comment:
I think there are a couple of spelling error in the comment; `clsters` and `defualt`. Not exactly super critical of course, but better to get it right, and some linters may also report it.

Also I'm not sure if our linter, when we run the workflows on this PR, would ask you to add an empty line above this section. Maybe you could just do that sooner rather than later.